roles/rsyslog/tasks/main.yaml

```diff
@@ -1,19 +1,19 @@
 ---
+- name: Get required packages
+  set_fact: _rsyslog_packages="{{ (rsyslog_base_packages) + (rsyslog_tls_packages if rsyslog_pki else []) + (rsyslog_additional_packages) }}"
+
 - name: Install/Update required packages
   package:
-    name: "{{ lookup('flattened', rsyslog_base_packages, rsyslog_tls_packages if rsyslog_pki else [], rsyslog_packages) }}"
+    name: "{{ item }}"
     state: latest
+  loop: "{{ _rsyslog_packages | flatten | d('') }}"
   when:
     - rsyslog_enabled | bool
     - not use_rsyslog_image | default(false) | bool
-  ignore_errors: yes
-  tags:
-    - skip_ansible_lint
+    - not item in ansible_facts.packages
 
 - name: Get rsyslog version
-  package:
-    list: rsyslog
-  register: rsyslog_version
+  set_fact: rsyslog_version=ansible_facts.packages['rsyslog'][0].version | default('') | int
```

Comment:
Doesn't the block below need adjusting? Also, does it need to be a fact instead of a variable local to the block?

Reply:
Fixed.